src/graph.h

```diff
@@ -113,9 +122,19 @@ struct Node {
   /// Possible values of mtime_:
   ///   -1: file hasn't been examined
   ///   0:  we looked, and file doesn't exist
-  ///   >0: actual file's mtime
+  ///   >0: actual file's mtime, or the latest mtime of its dependencies if it doesn't exist
```

Comment:
On the previous review, there was [comment](https://github.com/ninja-build/ninja/pull/1021#discussion_r307730374) expressing concern about changing the semantics of mtime(). Here are the instances where it is called in non-test cases outside of DependencyScan:

* [Builder::Cleanup](https://github.com/ninja-build/ninja/blob/master/src/build.cc#L535) - mtime() is used to determine whether we need to remove an output file when the build is interrupted. This change would produce a false positive here, but the ramifications would be that we call RemoveFile on an output that doesn't exist. 
* [Builder::FinishCommand](https://github.com/ninja-build/ninja/blob/master/src/build.cc#L760) - As far as I can tell, this function does not get called for phony edges, as there is no associated command.
* [Plan::CleanNode](https://github.com/ninja-build/ninja/blob/master/src/build.cc#L276) - My understanding is that this method is working its way back up the graph, marking nodes as clean after a command has been completed without changes. The mtime() call is being used to calculate the most recent input, which I believe would be fine.
* [ImplicitDepLoader::LoadDepsFromLog](https://github.com/ninja-build/ninja/blob/master/src/graph.cc#L627) - I don't believe this code path would be reached for phony edges.